libs/mutantschemata/CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+# vim: filetype=cmake
+
+# the C++ library
+set(dextool_mutantschemata_cpp_source_SRC
+    ${CMAKE_CURRENT_LIST_DIR}/cpp_source/api.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/cpp_source/type.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/cpp_source/cpp_string.cpp
+)
+add_library(dextool_mutantschemata_cpp_source STATIC ${dextool_mutantschemata_cpp_source_SRC})
+set_target_properties(dextool_mutantschemata_cpp_source PROPERTIES
+    COMPILE_FLAGS "${LIBLLVM_CXX_FLAGS} -O2 -g -I${CMAKE_CURRENT_LIST_DIR}/cpp_source"
+    ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/
+    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/
+    )
+# add clang libs
+target_link_libraries(dextool_mutantschemata_cpp_source
+        clangAST
+        clangASTMatchers
+        clangAnalysis
+        clangBasic
+        clangDriver
+        clangEdit
+        clangFrontend
+        clangFrontendTool
+        clangLex
+        clangParse
+        clangSema
+        clangEdit
+        clangRewrite
+        clangRewriteFrontend
+        clangStaticAnalyzerFrontend
+        clangStaticAnalyzerCheckers
+        clangStaticAnalyzerCore
+        clangIndex
+        clangSerialization
+        clangToolingCore
+        clangTooling
+        clangFormat
+)
+
+### The D-library
+file(GLOB_RECURSE SRC_FILES ${CMAKE_CURRENT_LIST_DIR}/source/mutantschemata/*.d)
+set(flags "-I${CMAKE_CURRENT_LIST_DIR}/source
+    -I${CMAKE_SOURCE_DIR}/source
+    -I${CMAKE_SOURCE_DIR}/clang
+    -I${CMAKE_SOURCE_DIR}/libclang
+    -I${CMAKE_SOURCE_DIR}/plugin/mutate/source
+    -I${CMAKE_SOURCE_DIR}/vendor/d2sqlite3/source
+    -I${CMAKE_SOURCE_DIR}/vendor/sumtype/src
+    -I${CMAKE_SOURCE_DIR}/vendor/miniorm/source
+    -I${CMAKE_SOURCE_DIR}/dextool_clang_extensions/source
+    -I${CMAKE_SOURCE_DIR}/vendor/cachetools/source
+    -I${CMAKE_SOURCE_DIR}/vendor/arsd"
+)
+
+set(deps "dextool_d2sqlite3;dextool_sumtype;dextool_miniorm;dextool_mutantschemata_cpp_source")
```

Comment:
move the `dextool_mutantschemata_cpp_source` to the beginning of this dependency list. Otherwise you may end up with hard to find linking errors.